frontend/src/DataImporter/createChannels.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  Channel,
+  Edge
+} from "./buildNetwork";
+import {
+  EdgeSpectrumDataRow
+} from "./parseCsv"
+
+export interface ChannelEdge {
+  id: string;
+  channel_label: string;
+  edges: string[];
+  frequency: number;
+  width: number;
+}
+
+
+/**
+ * Remove rows which ids are not in the edges list
+ *
+ *
+ * Check if, each edge from EdgeSpectrumData exists in the edges list
+ * An edge is removed from spectrum data if no mach is found in edges
+ * Should be called after MergeEdges
+ *
+ * @param edgeSpectrumData - EdgeSpectrumDataRow list
+ * @param edges - non redundant Edge list
+ * @returns - non redundant EdgeSpectrumDataRow list
+*/
+export const removeRedundantSpectrumRows = (edgeSpectrumData: EdgeSpectrumDataRow[], edges: Edge[]):EdgeSpectrumDataRow[] =>{
+  const channelMerged: EdgeSpectrumDataRow[] = []
+  const edgeIDs: string[] = edges.map(element => element.id)
+  for (const channel of edgeSpectrumData) {
+    if (edgeIDs.includes(channel.edgeId)) {
+      channelMerged.push(channel)
+    }
+  }
+  return channelMerged
+}
+
+/**
+ * Support function for groupSpectrumByChannel
+ * this function reads channel id from EdgeSpectrumDataRow
+ * and checks if Channel with that id has already been created
+ *
+ * if yes - append edge id to its list of edges
+ * if no - create new a Channel with 1 edge and append it to the list of Channels
+ *
+ * @param channelData - single EdgeSpectrumDataRow row
+ * @param channels - a list of all created Channels
+ * @returns updated ChannelEdge list
+*/
+export const getChannel = (channelData: EdgeSpectrumDataRow, channels: ChannelEdge[]): ChannelEdge[] => {
+  const cur_id = channelData.channelId;
+  const found = channels.find((channel) => channel.id == cur_id)
+  if (found) {
+    found.edges.push(channelData.edgeId);
+  } else {
+    const channel: ChannelEdge = {
+      id: channelData.channelId,
+      channel_label: channelData.channel_label,
+      edges: [channelData.edgeId],
+      frequency: channelData.frequency,
+      width: channelData.channelWidth
+    }
+    channels.push(channel)
+  }
+  return channels
+}
+
+/**
+ * Reads through parsed csv data and creates channel objects from it
+ *
+ * @param channelData - all EdgeSpectrumDataRow rows
+ * @returns - list of channel objects
```

Comment:
ewidentnie nie zwraca listy obiektów Channel, bo zwraca listę ChannelEdge, mylący ten opis jest. Coś z tymi nazwami trzeba lepiej ogarnąć moim zdaniem bo channel, channel edge, to się robi mylące.